### PR TITLE
CLI: Fix CRA init, ensure the new version of the preset

### LINKED ISSUE #21004: Fresh CRA fails in npm8 on beta.44
When running npx sb@next init in a brand-new CRA app created with the sandbox script (npx create-react-app . --template typescript), using npm 8.12.1, I get an error and installation does not complete.

```
Adding Storybook support to your "Create React App" based project(⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂) ⠼ reify: timing arbo
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: 7.0.0-beta.44@0.1.0
npm ERR! Found: @storybook/react@7.0.0-beta.44
npm ERR! node_modules/@storybook/react
npm ERR!   dev @storybook/react@"^7.0.0-beta.44" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer @storybook/react@">=5.2" from @storybook/preset-create-react-app@4.1.2
npm ERR! node_modules/@storybook/preset-create-react-app
npm ERR!   dev @storybook/preset-create-react-app@"^4.1.2" from the root project
```

Adding legacy-peer-deps = true to a .npmrc works around the issue.

https://discord.com/channels/486522875931656193/1072720859564883978/1072720859564883978

### PR DESCRIPTION
**Pull request by ndelangen**
Closes #21004

## What I did

I took a deeper look at the code, and it seemed like the reason it worked on my local machine is because our code that adds the preset, just takes thge latest version. When running verdaccio, the latest version IS the only version that's published..

But of course on the real npm there, the beta gets published as a beta, and thus isn't "latest".

I changed our code in the CLI so it takes the version from the versions file.

### CODE REVIEW

File: code/lib/cli/src/generators/REACT_SCRIPTS/index.ts [hunk 1]
```diff
@@ -38,8 +39,8 @@ const generator: Generator = async (packageManager, npmOptions, options) => {
     extraPackages.push('prop-types');
   }
 
-  // preset v3 is compat with older versions of CRA, otherwise let version float
-  const extraAddons = [`@storybook/preset-create-react-app${isCra5 ? '' : '@3'}`];
+  const version = versions['@storybook/preset-create-react-app'];
```
Comment (IanVS) [inline code/lib/cli/src/generators/REACT_SCRIPTS/index.ts:42]: Is this version compatible with all versions of CRA?  Or are we only supporting the latest CRA now?  If so, should we throw an error or something if a user tries to add SB 7 to an old CRA app?

Comment (ndelangen) [reply code/lib/cli/src/generators/REACT_SCRIPTS/index.ts:42]: I can throw an error is the version is below v5, yes. I think that's the right thing to do.
 
 I don't want to do active support for CRA <5 and storybook 7>

**Comment by IanVS** [comment]
Thanks for adding the check.  Do you think we also need a note in `MIGRATION.md`?  Or some other way to document it?  I'm not certain.

**Comment by ndelangen** [comment]
@shilman WDYT, a migration note required for dropping CRA < 5 support?

**Comment by shilman** [comment]
Good idea @ndelangen

**Review by shilman — APPROVED** [review]
LGTM!

**Pull request merged**